test/cases/compile_errors/add_on_undefined_value.zig

```diff
@@ -1,10 +1,24 @@
 comptime {
-    const a: i64 = undefined;
-    _ = a + a;
+    const undef: i64 = undefined;
+    const not_undef: i64 = 32;
+
+    // If either of the operands are zero, then the other operand is returned.
+    @compileLog(undef + 0);
+    @compileLog(not_undef + 0);
+    @compileLog(0 + undef);
+    @compileLog(0 + not_undef);
+
+    _ = undef + undef;
 }
 
 // error
 // backend=stage2
 // target=native
 //
-// :3:13: error: use of undefined value here causes undefined behavior
+// :11:17: error: use of undefined value here causes undefined behavior
+//
+// Compile Log Output:
+// @as(i64, undefined)
+// @as(i64, 32)
+// @as(i64, undefined)
+// @as(i64, 32)
```

Comment:
I'm pretty sure the testing system is currently ignoring this data. Does it fail if you change these lines?

I see the problem you are facing, however. You can't really put a check in for `== undefined` because that's not meaningful in the language.

To unblock you, here is my suggestion:
1. file an issue for adding support to the test cases framework to support checking these Compile Log Output lines.
2. manually verify the compile log output from the test cases from this PR (I assume you did this already)
3. reference the new issue in the commit or PR description

Then we can proceed with merging this PR.

Reply:
> Does it fail if you change these lines?

Yes, if I change one of my test cases and re-run `zig build test-cases`, the test will fail like this:
```zig
========= expected: =====================
zig-cache/o/df1ccd934478bbf9fa869378853117b0/tmp.zig:6:5: error: found compile log statement

Compile Log Output:
@as(i64, 32)
@as(i64, 32)
@as(i64, undefined)
@as(i64, 32)
@as(i64, undefined)
@as(i64, 33)
@as(i64, undefined)
@as(i64, 33)

========= but found: ====================
zig-cache/o/df1ccd934478bbf9fa869378853117b0/tmp.zig:6:5: error: found compile log statement

Compile Log Output:
@as(i64, undefined)
@as(i64, 32)
@as(i64, undefined)
@as(i64, 32)
@as(i64, undefined)
@as(i64, 33)
@as(i64, undefined)
@as(i64, 33)
```

I did see other test cases that used the compile log output for testing so I assumed they were being checked. While writing the new test cases, I often had to edit my test cases because I had some mistakes in them that I only realized after running `zig build test-cases` again.

Reply:
Huh, never mind all that other stuff I said then!